packages/ethers/src/index.ts

```diff
@@ -128,13 +128,21 @@ export class TurnkeySigner extends AbstractSigner implements ethers.Signer {
         activityType: type,
       });
     } else {
-      const activity = await this.client.signTransaction({
+      const { activity, signedTransaction } = await this.client.signTransaction({
         signWith: this.signWith,
         type: "TRANSACTION_TYPE_ETHEREUM",
-        unsignedTransaction: message,
+        unsignedTransaction,
       });
 
-      return assertNonNull(activity?.signedTransaction);
+      if (activity.status !== "ACTIVITY_STATUS_COMPLETED") {
```

Comment:
What exactly does this look like in the case where an activity requires consensus? _Should_ an error be thrown?

Reply:
Great question — ended up having the default behavior be to poll for 3 seconds, then return an error. At that point, it's up to the end-user to figure out how they want to handle the error. I've included some examples of handling in the various `retry.ts` files